src/content/1.7/development/architecture/modern/controller-routing.md

```diff
@@ -0,0 +1,314 @@
+Same applies to **Improve**,  **Sell** sections etc.
+
+This is how the directory tree of controllers should look like:
+
+```
+Controller/
+└── Admin
+    ├── Configure
+    │   ├── AdvancedParameters
+    │   └── ShopParameters
+    ├── Improve
+    │   ├── Design
+    │   ├── International
+    │   ├── Modules
+    │   ├── Payment
+    │   └── Shipping
+    └── Sell
+        ├── Catalog
+        ├── Customers
+        ├── CustomerService
+        ├── Orders
+        └── Stats
+```
+Symfony Controllers should be thin by default and have only one responsibility: getting the HTTP Request from the client and returning an HTTP Response. This means that every business logic should be placed in dedicated classes outside the Controller:
+
+* Form management
+* Database access
+* Validation
+* etc...
+
+You can take a look at [PerformanceController](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/PerformanceController.php) for an example of good implementation, and [ProductController](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/ProductController.php) for something you should avoid at all costs.
+
+{{% notice warning %}}
+**Never, ever call the legacy controller inside the new controller**. It's a no go, no matter the reason!
+{{% /notice %}}
+
+Controllers are responsible for performing "Actions". Actions are methods of Controllers which mapped to a route, and that return a `Response`.
+
+{{% notice tip %}}
+**Try to avoid creating helper methods in your controller.** If you find yourself needing them, it could be a symptom of your Controller becoming too complex. This can be solved by extracting code into external classes as needed.
+{{% /notice %}}
+
+Regarding the rendering of a Response, there is some data specific to PrestaShop (in Back Office) that we must set for every action:
+
+| Attribute                   |  Type                          |  Description                                            |
+|-----------------------------|--------------------------------|---------------------------------------------------------|
+| `layoutHeaderToolbarBtn`    | [['href', 'des','icon'], ...]  | Set buttons in toolbar on top of the page               |
+| `layoutTitle`               | string                         | Main title of the page                                  |
+| `requireAddonsSearch`       | boolean                        | If *true*, display addons recommendations button        |
+| `requireBulkActions`        | boolean                        | If *true*, display bulk actions button                  |
+| `showContentHeader`         | boolean                        | If *true*, display the page header                      |
+| `enableSidebar`             | boolean                        | If *true*, display a sidebar                            |
```

Comment:
Do we know if this sidebar is the left sidebar that contains the menu navbar ; or if this sidebar is the right sidebar that opens when we click `Help` button, or something else?